### PR TITLE
Improve maintainability of Case Studies styling

### PR DESCRIPTION
**Pull request by habibrosyad**
This is an attempt to improve the maintainability of Case Studies as outlined in #23660 by:
- Adding quote and lead shortcode
- Adding case study metadata in front matter for rendering common contents (e.g. heading)
- ~~Use markdown for content format (reducing the use of HTML and CSS styles)~~
- Allow case study page to inherit similar styles from a centralised CSS file. At the moment all styles are placed in `new-case-study.css`. It is temporary. It is done so to avoid breaking changes, as not all case studies (especially in languages other than English) have been updated with this change.
- Make font sizes & weights more consistent with the rest of the site.

~~The changes cover 3 case studies (otherwise it's might be to huge to review for an initial attempt):~~
- ~~https://k8s.io/case-studies/adform/~~
- ~~https://k8s.io/case-studies/adidas/~~
- ~~https://k8s.io/case-studies/amadeus/~~

This PR has been updated to cover styles of all case studies (English)

/area web-development

### CODE REVIEW

File: layouts/case-studies/single.html [hunk 1]
```diff
@@ -2,6 +2,37 @@
 {{ with .Params.content_url }}
 <meta http-equiv="refresh" content="0; URL='{{ . }}'" />
 {{ else }}
-	{{ .Content }}
+	{{ if .Params.new_case_study_styles }}
```
Comment (habibrosyad) [inline layouts/case-studies/single.html:5]: This param is used for backward compatibility with the current styles. Can be removed later (in other PR) once all case study pages has been update.

**Comment by habibrosyad** [comment]
Preview:
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/adform/
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/adidas/
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/amadeus/

**Review by sftim — COMMENTED** [review]
@habibrosyad thanks for this PR

I recommend separating out two kind of change, roughly as:
- pull requests that switch one or more case studies from HTML to Markdown, leaving the rendering equivalent
- pull requests that change the styling for all case studies, or perhaps just for all Markdown-based case studies

This separation will make the site look more consistent during a transition - not just for English, but also for all the downstream localizations.

If you compare https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/appdirect/ to https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/adform/ then you will see differences; at least, I do.

Are you convinced? Issue #23660 doesn't mention switching to Markdown and so, whilst I do like the idea of switching, I think it makes sense to have the migration to Markdown not have any visual impact.

**Comment by habibrosyad** [comment]
@sftim thanks for the feedback

> If you compare https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/appdirect/ to https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/adform/ then you will see differences; at least, I do.

I do expect this, since the styles are indeed different. It was just that I'm not so sure to apply the changes to all those pages before getting some feedbacks here.

I agree this should be separated into several PRs as you suggested. So I'm thinking to take this approach:
1.  Update this PR to focus on updating the styling for all case studies, the options are:
    - Updating all case studies at once (a PR/language, so this PR will focus only on English)
    - A PR for a case study

    @sftim  which options do you think is better? IMHO the first option would be better for a smooth transition as the later is just what this PR is (only made changes for 3 pages, while the rests looks different like what you've noticed).
2. After all styles updated, starts the transition to markdown. These PRs should not have any visual impact.

**Comment by sftim** [comment]
> I'm thinking to take this approach:
> 1. Update this PR to focus on styling, the options are:
>      - Updating all case studies at once (a PR/language, so this PR will focus only on English based case studies)
>      - A separate PR for a case study
>
>    @sftim which options do you think is better? IMHO the first option would be better for a smooth transition as the later is just what this PR is (only made changes for 3 pages, while the rests looks different like what you've noticed).
>
> 2. After all styles updated, starts the transition to markdown. These PRs should not have any visual impact.

That order of work looks OK to me. It's also OK to propose changing the styles for _all_ the case studies. Once this PR is redone to focus on CSS and styles, we'll `/label area/web-development` it as well, to highlight what kind of change it is.

**Comment by habibrosyad** [comment]
All case studies has been updated, could you review again @sftim? 

The odd one is https://k8s.io/case-studies/wikimedia/ which updated into https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/wikimedia/

**Comment by sftim** [comment]
This change will look odd during localization: some case studies will look different to others.

I'd be much more keen to see a change where we make the CSS more maintainable _without_ that jump in styling between different case studies.

**Comment by habibrosyad** [comment]
> I'd be much more keen to see a change where we make the CSS more maintainable _without_ that jump in styling between different case studies.

@sftim Do you mean case studies between different localization (which hasn't been updated yet)? 

This change reduces the markup to the bare minimum for easier maintenance and transition to markdown later (only basic H2, H4, and Ps + some shortcodes) as well as the overall styles (yeah this part introduces that jump in style). 

If the issue is with the totally different looks (color, font size, etc), do you prefer we retain the old style and focus on the maintenance of markups and CSS (without changing the current appearance of the pages)?

**Comment by sftim** [comment]
Imagine you're a localization team and, after this merges, you only have enough time to update a few case studies, not all of them. That's not ideal because after you update the first few case studies, your localization gets a mix of appearances.

So the first change could make the styles be more maintainable without revising appearance. Then, after localization teams have had a chance to update, revise the appearance of case study pages to facilitate a later 3rd switch: replace HTML with Markdown.

Only the 2nd change would affect appearance of case studies; the others are enabling work.

**Comment by habibrosyad** [comment]
> Imagine you're a localization team and, after this merges, you only have enough time to update a few case studies, not all of them. That's not ideal because after you update the first few case studies, your localization gets a mix of appearances.

This make sense. I will adjust this PR to match the existing styles and focus more on the maintainability of HTML tags and CSS.

**Comment by habibrosyad** [comment]
@sftim I've updated the PR to reduce the differences with the current styles (while also addressing #23660). Could you check again?

Some previews: 
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/zalando/
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/zh/case-studies/zalando/
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/case-studies/pearson/
- https://deploy-preview-23708--kubernetes-io-master-staging.netlify.app/ko/case-studies/pearson/

Aside from that, looking closely to the case studies, there are actually 3 different styles:
- With gradient (e.g https://k8s.io/case-studies/adidas/)
- Without gradient (most of case studies use this one, e.g https://k8s.io/case-studies/zalando)
- The only weird one https://k8s.io/case-studies/wikimedia (I don't think we want to keep this style as it is now)

And to compare between different localizations I found that some of them are broken or actually not yet translated despite being registered in the language menu, e.g:
- https://kubernetes.io/zh/case-studies/ygrene/ (broken layout)
- https://kubernetes.io/ko/case-studies/ygrene/ (not yet translated)
- https://kubernetes.io/zh/case-studies/chinaunicom/ (broken images because of #22636)

**Comment by celestehorgan** [comment]
@habibrosyad 

🎉 This is an awesome PR. Some notes before we begin review:

> And to compare between different localizations I found that some of them are broken or actually not yet translated despite being registered in the language menu, e.g:
> 
>    https://kubernetes.io/zh/case-studies/ygrene/ (broken layout)
>    https://kubernetes.io/ko/case-studies/ygrene/ (not yet translated)
>    https://kubernetes.io/zh/case-studies/chinaunicom/ (broken images because of #22636)

It might be a good idea to wait for translation teams to pull in changes from #22636 before we merge this. WDYT @sftim?

**Comment by habibrosyad** [comment]
> It might be a good idea to wait for translation teams to pull in changes from #22636 before we merge this. WDYT @sftim?

I think I could help with PRs for that as well if there are none yet...

**Comment by sftim** [comment]
> I think I could help with PRs for that as well if there are none yet...

@habibrosyad that would be great.
If you want, feel free to mention your plans in https://kubernetes.slack.com/messages/sig-docs-localizations

**Comment by habibrosyad** [comment]
> @habibrosyad that would be great.

Made PR https://github.com/kubernetes/website/pull/23806 for starter.

> If you want, feel free to mention your plans in https://kubernetes.slack.com/messages/sig-docs-localizations

Sure.

**Comment by sftim** [comment]
Are there any more PRs like https://github.com/kubernetes/website/pull/23806 coming? (Right now I don't have time to help much with that myself).

**Comment by habibrosyad** [comment]
> Are there any more PRs like #23806 coming? (Right now I don't have time to help much with that myself).

Sorry for the delay. I'm planning to finish PRs for the other localisations in this week. Raised one here #24400 for Japanese.

**Comment by habibrosyad** [comment]
Just created the last 2 PRs for #22636.

I found that it's slightly weird to pull the changes for Korean, because none of the case studies are localised. Same case also for some content in Chinese.

**Comment by sftim** [comment]
@habibrosyad _if you know how to_, please rebase this against the master branch. That will let Netlify build a current preview.

If you're not sure, don't worry - reviewers can manually simulate the same thing locally to verify these changes. Just let us know if that's a concern and we'll help out.

**Comment by habibrosyad** [comment]
@sftim rebased to the master branch.

**Comment by kbhawkey** [comment]
@celestehorgan @sftim, Do you have time to review these changes? Thanks!

**Comment by sftim** [comment]
@habibrosyad this does look good
/lgtm

**Review by irvifa — COMMENTED** [review]
Thanks
/approve

**Pull request merged**